plugins/floatingspace/plugin.js

```diff
@@ -282,16 +282,16 @@
 						'<div id="{topId}" class="cke_top" role="presentation">{content}</div>' +
 					'</div>' +
 				'</div>' ),
-				floatSpace = CKEDITOR.document.getBody().append( CKEDITOR.dom.element.createFromHtml( floatSpaceTpl.output( {
+				floatSpace = CKEDITOR.document.getBody().append(CKEDITOR.dom.element.createFromHtml(CKEDITOR.tools.htmlSafeByReview(floatSpaceTpl.output({
 					content: topHtml,
```

Comment:
how is `topHtml` generated?

Reply:
I think we get it from `editor.fire('uiSpace'` so let's try to trace what uiSpace does

Reply:
topHtml and bottomHtml are set to the output of firing uiSpace. The uiSpace description can be found here: https://sourcegraph.com/github.com/ckeditor/ckeditor4/-/blob/core/creators/themedui.js?L620-630. In essence, the uiSpace event is fired when a modification to the top or bottom bar needs to be made. This is triggered in a few places but for example the resize plugin makes changes as seen here: https://sourcegraph.com/github.com/ckeditor/ckeditor4/-/blob/plugins/resize/plugin.js?L80-103. Overall, topHtml and bottomHtml would be safe as long as each of these internal modifications do not pull in any user data. I'll review the usages and update the justification as necessary.